src/org/opensolaris/opengrok/history/FileHistoryCache.java

```diff
@@ -309,12 +303,45 @@ private void storeFile(History histNew, File file, Repository repo) throws Histo
                 }
             }
         } catch (IOException ex) {
-            // Ideally we would want to catch the case when incremental update
-            // is done but the cached file is not there however we do not have
-            // the data to do it here.
+            LOGGER.log(Level.SEVERE, "Cannot open history cache file " + cacheFile.getPath());
+        }
+
+        return history;
+    }
+
+    /**
+     * Store history object (encoded as XML and compressed with gzip) in a file.
+     *
+     * @param history history object to store
+     * @param file file to store the history object into
+     * @param repo repository for the file
+     * @param mergeHistory whether to merge the history with existing or
+     *                     store the histNew as is
+     * @throws HistoryException
+     */
+    private void storeFile(History histNew, File file, Repository repo,
+            boolean mergeHistory) throws HistoryException {
+
```

Comment:
I meant that if you do this:
```java
private void storeFile(History histNew, File file, Repository repo) {
    storeFile(histNew, file, repo, false);
}

private void storeFile(History histNew, File file, Repository repo,
            boolean mergeHistory) {
    // the code as you have it
}
```

Then you keep your constructs and allow the callee not to worry about renamed parameter.
This is still valid
```java
        if (!file.isDirectory()) {
            storeFile(hist, file, repository, !renamed);
        }
```
As well as this:

```java
// retrieving the history takes too long, cache it!
storeFile(history, file, repository);
```

This doesn't really bring a code duplication. However all of these are just cosmetics.